web/src/main/webapp/resources/js/atsy-settings.js

```diff
@@ -43,8 +43,13 @@ function SettingsForm() {
                     }
                     $this.find('input').val('');
                 }).error(function (xhr) {
-                    var response = xhr.responseJSON;
-                    showError($this, response.errorMessage);
+                    // If the response type is not JSON, redirection happened -> perform a manual page reload
+                    if ( xhr.status === 200 && xhr.getResponseHeader("Content-Type").indexOf("json") === -1 ) {
```

Comment:
Please check these documentations:
* http://api.jquery.com/ajaxError/
* http://stackoverflow.com/questions/16080462/handling-session-timeout-in-ajax

Using the `$(document).ajaxError()` event handler can prevent anybody to copy&paste the same code for each ajax request.